pkg/controller/portworxdiag/rbac.go

```diff
@@ -0,0 +1,121 @@
+package portworxdiag
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	serviceAccountName = "portworx-diag-collector"
+	roleName           = "diag-collector"
+	roleBindingName    = "diag-collector"
+)
+
+func objectExists(c client.Client, ref types.NamespacedName, emptyObj client.Object) (bool, error) {
+	err := c.Get(context.TODO(), ref, emptyObj)
+	if err == nil {
+		return true, nil
+	}
+	if errors.IsNotFound(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+func ensureRBACExists(client client.Client, ns string) error {
+	if err := createServiceAccountIfNotExist(client, ns); err != nil {
+		return err
+	}
+	if err := createRoleIfNotExist(client, ns); err != nil {
+		return err
+	}
+	if err := createRoleBindingIfNotExist(client, ns); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createServiceAccountIfNotExist(c client.Client, ns string) error {
```

Comment:
We already have utility methods to CreateOrUpdateXXX for different objects 
 in utils. - https://github.com/libopenstorage/operator/blob/master/pkg/util/k8s/k8s.go#L197

Eg usage - https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/component/portworx_basic.go#L136
All components use these method to create objects.

Reply:
Discussed offline, we will instead just use the Portworx SA and make sure it has permission to read/write all the portworx v1 objects we need